lib/hanami/model/sql.rb

```diff
@@ -0,0 +1,158 @@
+    # rollback them (down).
+    #
+    # We can use <tt>up</tt> and <tt>down</tt> blocks for custom strategies, or
+    # only one <tt>change</tt> block that automatically implements "down" strategy.
+    #
+    # @param blk [Proc] a block that defines up/down or change database migration
+    #
+    # @since 0.4.0
+    #
+    # @example Use up/down blocks
+    #   Hanami::Model.migration do
+    #     up do
+    #       create_table :books do
+    #         primary_key :id
+    #         column :book, String
+    #       end
+    #     end
+    #
+    #     down do
+    #       drop_table :books
+    #     end
+    #   end
+    #
+    # @example Use change block
+    #   Hanami::Model.migration do
+    #     change do
+    #       create_table :books do
+    #         primary_key :id
+    #         column :book, String
+    #       end
+    #     end
+    #
+    #     # DOWN strategy is automatically generated
+    #   end
+    def self.migration(&blk)
+      Migration.new(configuration.gateways[:default], &blk)
+    end
+
+    # SQL adapter
+    #
+    # @since x.x.x
+    module Sql
+      # Returns a SQL fragment that references a database function by the given name
+      # This is useful for database migrations
+      #
+      # @param name [String,Symbol] the function name
+      # @return [String] the SQL fragment
+      #
+      # @since x.x.x
+      #
+      # @example
+      #   Hanami::Model.migration do
+      #     up do
+      #       execute 'CREATE EXTENSION IF NOT EXISTS "uuid-ossp"'
+      #
+      #       create_table :source_files do
+      #         column :id, 'uuid', primary_key: true, default: Hanami::Model::Sql.function(:uuid_generate_v4)
+      #         # ...
+      #       end
+      #     end
+      #
+      #     down do
+      #       drop_table :source_files
+      #       execute 'DROP EXTENSION "uuid-ossp"'
+      #     end
+      #   end
+      def self.function(name)
+        Sequel.function(name)
+      end
+
+      # Returns a literal SQL fragment for the given SQL fragment.
+      # This is useful for database migrations
+      #
+      # @param string [String] the SQL fragment
+      # @return [String] the literal SQL fragment
+      #
+      # @since x.x.x
+      #
+      # @example
+      #   Hanami::Model.migration do
+      #     up do
+      #       execute %{
+      #         CREATE TYPE inventory_item AS (
+      #           name            text,
+      #           supplier_id     integer,
+      #           price           numeric
+      #         );
+      #       }
+      #
+      #       create_table :items do
+      #         column :item, 'inventory_item', default: Hanami::Model::Sql.literal("ROW('fuzzy dice', 42, 1.99)")
+      #         # ...
+      #       end
+      #     end
+      #
+      #     down do
+      #       drop_table :itmes
+      #       execute 'DROP TYPE inventory_item'
+      #     end
+      #   end
+      def self.literal(string)
```

Comment:
Fair enough, I guess my comment is only true of people experienced with Sequel anyway